core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -737,8 +734,8 @@ public void executeSystemTask(WorkflowSystemTask systemTask, String taskId, int
                     logger.warn("Concurrent Execution limited for {}:{}", taskId, task.getTaskDefName());
                     return;
                 }
-                if (task.getRateLimitPerSecond() > 0 && !rateLimitingService.evaluateRateLimitBoundary(task)) {
-                    logger.warn("RateLimit Execution limited for {}:{}, limit:{}", taskId, task.getTaskDefName(), task.getRateLimitPerSecond());
+                if (task.getRateLimitPerFrequency() > 0 && executionDAO.exceedsRateLimitPerFrequency(task)) {
+                    logger.warn("RateLimit Execution limited for {}:{}, limit:{}", taskId, task.getTaskDefName(), task.getRateLimitPerFrequency());
                     return;
```

Comment:
feels like collecting metrics here might be helpful if we can start getting a lot of task executions hitting this block.

Reply:
Metrics is collected in the `RedisMetadataDao`, collecting here would be redundant